src/OAuth/Credential.php

```diff
@@ -92,4 +92,20 @@ public function getRefreshToken(): string
     {
         return $this->refreshToken;
     }
+
+    /**
+     * @param string $accessToken
+     * @param string $refreshToken
+     *
+     * @return Credential
+     */
+    public function withRefreshedTokens(string $accessToken, string $refreshToken): self
```

Comment:
withNewCredentials ? (withRefreshedTokens sounds confusing as it implies there are multiple refresh tokens :p)